src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -84,8 +84,20 @@ private static boolean isValidPath(Path filePath) {
      * @throws StorageOperationException if there were errors converting and/or storing data to file.
      */
     public void save(AddressBook addressBook) throws StorageOperationException {
+    	int saveAttempts = 2;
+        save(addressBook, 2);
+    }
 
-        /* Note: Note the 'try with resource' statement below.
+    /**
+     * internal method to save address book to file. 
+     * takes in an addressbook and the number of save attempts left to try.
+     * tries to save the addressBook to the file (given by path) attemptsLeft times
+     * @param addressBook
+     * @param attemptsLeft
+     * @throws StorageOperationException
```

Comment:
Need to give a description of the param or the exception. If you have nothing more to say, then you can remove the line altogether.